### ISSUE TITLE
Porting app to MacOS (macatalyst) mouse drag, zoom, and context menu not working.

### ISSUE THREAD

**Issue by King-Taz** [ScottPlot/ScottPlot#4572]
**Issue:** I'm porting a Maui NET8 app developed in Windows using VS to MacOS. Everything works well except mouse drag, zoom, and context menu. Please note, I'm simply copying the VS folder to my Mac, opening it in VS Code and compiling/debugging it. This has worked with other apps but may not work with ScottPlot.

**ScottPlot Version:** 5.0.47 

**Code Sample:** 

XAML code
```cs
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:ScottPlot="clr-namespace:ScottPlot.Maui;assembly=ScottPlot.Maui"
             x:Class="PlottingWithScott.MainPage">

    <Grid ColumnDefinitions="*" RowDefinitions="*" Padding="0,0,0,0" >
        <ScottPlot:MauiPlot x:Name="MauiPlot1" Scale="1" />
    </Grid>
</ContentPage>
```
and calling this method after ```InitializeComponent();``` in the code behind
```
        private void SimpleDrawPhasorPlot()
        {
            //Create arrow base coordinates used for all vectors
            Coordinates arrowBase = new(0, 0);

            // create arrow 1
            Coordinates arrow1Tip = new(5, 0);
            CoordinateLine arrow1Line = new(arrowBase, arrow1Tip);

            // add a simple arrow
            var arrow1 = MauiPlot1.Plot.Add.Arrow(arrow1Line);
            arrow1.ArrowFillColor = Colors.Red;
            arrow1.ArrowLineWidth = 0;
            arrow1.ArrowWidth = 5;
            arrow1.ArrowheadLength = 20;
            arrow1.ArrowheadAxisLength = 20;
            arrow1.ArrowheadWidth = 20;
            var dl1 = MauiPlot1.Plot.Add.Text("T(1)", arrow1Tip);
            dl1.LabelFontSize = 28;
            dl1.LabelFontColor = Colors.Black;

            // Preserve X-axis plot scaling during window size changes
            SquarePreserveX rule2 = new(xAxis: MauiPlot1.Plot.Axes.Bottom, yAxis: MauiPlot1.Plot.Axes.Left);
            MauiPlot1.Plot.Axes.Rules.Clear();
            MauiPlot1.Plot.Axes.Rules.Add(rule2);

        }
```

**Comment by KosmosWerner** [comment]
In the `MauiPlot.cs` file, the following lines of code can be found:

```csharp
if (DeviceInfo.Idiom == DeviceIdiom.Desktop)
{
    Touch += MauiPlot_Touch;
    EnableTouchEvents = true;
}
else
{
    EnableTouchEvents = false;
    var panGestureRecognizer = new PanGestureRecognizer();
    ...
}
```

This was implemented because click events could register more than twice if not distinguished by platform. However, it seems that `MacCatalyst` is not recognized as `Desktop`.  

Instead, you can create a class that inherits from `MauiPlot` as follows:  

```csharp
using ScottPlot;
using ScottPlot.Interactivity;
using ScottPlot.Maui;
using SkiaSharp.Views.Maui;
using System.Diagnostics;

namespace YourNamespace; 

internal class CustomMauiPlot : MauiPlot
{
    public CustomMauiPlot() : base()
    {
        if (DeviceInfo.Platform == DevicePlatform.MacCatalyst)
        {
            Touch += MauiPlot_Touch;
            EnableTouchEvents = true;
        }
    }

    private void MauiPlot_Touch(object? sender, SKTouchEventArgs e)
    {
        Pixel pixel = new Pixel(e.Location.X, e.Location.Y);
        switch (e.ActionType)
        {
            case SKTouchAction.Pressed:
                {
                    IUserAction action = e.MouseButton switch
                    {
                        SKMouseButton.Left => new ScottPlot.Interactivity.UserActions.LeftMouseDown(pixel),
                        SKMouseButton.Middle => new ScottPlot.Interactivity.UserActions.MiddleMouseDown(pixel),
                        SKMouseButton.Right => new ScottPlot.Interactivity.UserActions.RightMouseDown(pixel),
                        _ => new ScottPlot.Interactivity.UserActions.Unknown()
                    };
                    UserInputProcessor.Process(action);

                    break;
                }
            case SKTouchAction.Moved:
                {
                    IUserAction action = new ScottPlot.Interactivity.UserActions.MouseMove(pixel);
                    UserInputProcessor.Process(action);

                    break;
                }
            case SKTouchAction.Released:
                {
                    IUserAction action = e.MouseButton switch
                    {
                        SKMouseButton.Left => new ScottPlot.Interactivity.UserActions.LeftMouseUp(pixel),
                        SKMouseButton.Middle => new ScottPlot.Interactivity.UserActions.MiddleMouseUp(pixel),
                        SKMouseButton.Right => new ScottPlot.Interactivity.UserActions.RightMouseUp(pixel),
                        _ => new ScottPlot.Interactivity.UserActions.Unknown()
                    };
                    UserInputProcessor.Process(action);
                    break;
                }
            case SKTouchAction.WheelChanged:
                {
                    IUserAction action = e.WheelDelta > 0
                      ? new ScottPlot.Interactivity.UserActions.MouseWheelUp(pixel)
                      : new ScottPlot.Interactivity.UserActions.MouseWheelDown(pixel);

                    UserInputProcessor.Process(action);
                    break;
                }
            default: break;
        }
    }
}
```

Use this class instead of `MauiPlot`.

**Comment by King-Taz** [comment]
> However, it seems that MacCatalyst is not recognized as Desktop.

Wouldn't this be considered a Maui bug? Because I would expect Maui to do the proper input translation based on the idiom. Or is that wrong?

**Comment by King-Taz** [comment]
I ask because if this is a Maui bug, I will submit it as such.  NET Maui is not very cross-platform if it doesn't translate mouse/keyboard properly based on  idiom IMO.

Also, will this class be incorporated into ScottPlot or will developers need to create it for themselves?

**Comment by KosmosWerner** [comment]
Sorry for the late response. This should be a bug in MAUI, as it should recognize that it is running on a desktop device. For now, the appropriate solution would be to use the approach I provided by creating a derived class, as I am unaware of the behavior on Apple devices, and there may be other particular behaviors on those platforms.

Additionally, MAUI fails to recognize the keyboard properly. Even to make the window respond to the mouse wheel, I had to use events from a SkiaSharp canvas, as MAUI does not handle them correctly.

**Comment by bclehmann** [comment]
> I ask because if this is a Maui bug, I will submit it as such. NET Maui is not very cross-platform if it doesn't translate mouse/keyboard properly based on idiom IMO.

I don't think this is a Maui bug, since Microsoft seems to know about it. They have this article on it here: https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/user-interface-idiom?view=net-maui-9.0

TLDR: Mac Catalyst apps default to the iPad UI idiom, you can change this by modifying your `Info.plist` file. It looks like it comes with some downsides, as some UI controls seem to only be fully supported in one idiom or the other. I think this default behaviour is because Mac Catalyst was created for moving iPad apps over to Mac, it's just unfortunate since Mac Catalyst is the only supported Maui platform on MacOS, likely stemming from Maui's roots as a mobile platform.

If you want to still use iPad UI idiom then you'll have to override MauiPlot's idiom-specific behaviour.

**Comment by swharden** [comment]
Thanks for reporting this issue, and to those of you who gave insightful feedback as well!

I think the solution to the original question was described in https://github.com/ScottPlot/ScottPlot/issues/4572#issuecomment-2507033826 so I'll close this issue, but I appreciate that cross-platform UI consistency is a hard problem that even .NET MAUI struggles to solve 😅
